craft_cli/messages.py

```diff
@@ -571,6 +571,27 @@ def verbose(self, text: str) -> None:
             use_timestamp = True
         self._printer.show(stream, text, use_timestamp=use_timestamp)
 
+    @_active_guard()
+    def warning(self, text: str, *, prefix: str = "Warning: ") -> None:
+        """Show an important warning to the user.
+
+        To show warnings to the user, which are not errors but may
+        indicate that something is not right and the user should pay attention to it.
+
+        :param prefix: Display text prepended to warnings, defaults to "Warning: "
+        """
+        text = f"{prefix}{text}"
+        if self._mode == EmitterMode.QUIET:
```

Comment:
I think this would look nicer with a [match statement](https://docs.python.org/3/tutorial/controlflow.html#match-statements), but that's not a requirement at all.